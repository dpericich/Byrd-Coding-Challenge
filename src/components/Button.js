import React from "react";
import "./button.css";

const Button = ({ message = "", color = "", confirmSubmitForm }) => {
  return (
    <div className={`button ${color}`} onClick={confirmSubmitForm}>
      {message}
    </div>
  );
};

export default Button;
