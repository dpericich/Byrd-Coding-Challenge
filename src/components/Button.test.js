import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Button from "./Button";

describe("<Button />", () => {
  test("it should render correctly", () => {
    const component = shallow(<Button />);

    expect(toJSON(component)).toMatchSnapshot();
  });
  test("it should render button with correct string text and color", () => {
    const string = "Submit";
    const color = "btn-orange";

    const component = shallow(<Button message={string} color={color} />);

    expect(toJSON(component)).toMatchSnapshot();
  });
});
