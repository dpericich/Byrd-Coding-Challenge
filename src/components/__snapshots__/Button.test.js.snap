// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it should render button with correct string text and color 1`] = `
<div
  className="button btn-orange"
>
  Submit
</div>
`;

exports[`<Button /> it should render correctly 1`] = `
<div
  className="button "
/>
`;
